src/NServiceBus.AcceptanceTests/Basic/When_starting_up_the_endpoint.cs

```diff
@@ -0,0 +1,84 @@
+﻿namespace NServiceBus.AcceptanceTests.Basic
+{
+    using System;
+    using System.Linq;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_starting_up_the_endpoint : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public void Should_log_warning_if_queue_is_configured_with_anon_and_everyone_permissions()
+        {
+            var context = new Context
+            {
+                Id = Guid.NewGuid()
+            };
+            Scenario.Define(context)
+                .WithEndpoint<EndPoint>(b => b.Given((bus, c) => bus.SendLocal(new MyMessage
+                {
+                    Id = c.Id
+                })))
+                .Done(c => c.WasCalled)
+                .Run();
+
+            var logItem = context.Logs.FirstOrDefault(item => item.Message.Contains(@"[Everyone] and [NT AUTHORITY\ANONYMOUS LOGON]"));
+            Assert.IsNotNull(logItem);
+            StringAssert.Contains(@"is running with [Everyone] and [NT AUTHORITY\ANONYMOUS LOGON] permissions. Consider setting appropriate permissions, if required by your organization", logItem.Message);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool WasCalled { get; set; }
+            public Guid Id { get; set; }
+
+            public bool GeneratorWasCalled { get; set; }
+        }
+
+        public class EndPoint : EndpointConfigurationBuilder, IWantToRunBeforeConfigurationIsFinalized
+        {
+            static bool initialized;
+            public EndPoint()
+            {
+                if (initialized)
+                {
+                    return;
+                }
+                initialized = true;
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.UseTransport<MsmqTransport>();
+                });
+            }
+
+            static Context Context { get; set; }
```

Comment:
@danielmarbach honestly, I used another test as a basis for this, because when I tried to do it by hand (e.g. without sending an actual message, or defining a handler), the test wouldn't finish. So I just used clipboard inheritance to fix it :)